enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -88,8 +88,10 @@ public JWTAuthenticator() {
     @Override
     public boolean canAuthenticate(RequestContext requestContext) {
         if (isJWTEnabled(requestContext)) {
-            String jwt = retrieveAuthHeaderValue(requestContext);
-            return jwt != null && jwt.split("\\.").length == 3;
+            String authHeaderValue = retrieveAuthHeaderValue(requestContext);
+            return authHeaderValue != null &&
+                    authHeaderValue.trim().split("\\s+").length == 2 &&
```

Comment:
this makes `canAuthenticate` true for invalid tokens (ex: token with space in the middle).
Isn't is be a problem?

Reply:
Yes. This only checks the format. The validity is checked in the authenticate method. Regarding example mentioned, I guess we can check the `Bearer` keyword too.

Reply:
Seems there is a scenario where we must check the keyword `bearer` in the `canAuthenticate` method itself. This would prevent conflicts with custom authentication (that maybe added with custom filters / interceptors / opa) which also includes a JWT in the auth header, yet with a scheme other than "bearer".